src/components/UserInfo/UserInfo.tsx

```diff
@@ -1 +1,11 @@
-export const UserInfo = () => {};
+import { User } from '../../types/User';
+
+type Props = {
+  user: User | undefined;
+};
+
+export const UserInfo: React.FC<Props> = ({ user }) => (
+  <a className="UserInfo" href={user ? `mailto:${user.email}` : '#'}>
+    {user ? user.name : 'Unknown user'}
+  </a>
+);
```

Comment:
it is better to handle undefined case in parent component
User should always accept User type as a prop

in parent component you can use just conditional rendering